packages/react/src/internal/FeatureFlags.js

```diff
@@ -17,4 +17,20 @@
  * const MyComponent = props => (<div {...props}>{aFeatureFlag ? 'foo' : 'bar'}</div>);
  */
 
-/* Currently no feature flag is in use, but keeping this file as a placeholder */
+/**
+ * Uses `value` prop to turn several components to controlled mode, notablly `<NumberInput>`.
+ * The "controlled mode" detection is done in component creation time and _never_ changes after creation.
+ * To ensure the component is in controlled mode, set non-`undefined` value at creation time.
+ *
+ * This flag also disables prop -> state sync in several components, notablly `<NumberInput>`.
+ *
+ * Also, this flag ensures that new value upon user gesture is sent via event handlers,
+ * so appilcation can set the latest value to control our components.
+ *
+ * * _With_ this feature flag, the event handler has `onChange(event, { value, direction })` where:
+ *   * `event` is the (React) raw event
+ *   * `value` is the new value
+ *   * `direction` tells you the button you hit is up button or down button
```

Comment:
The `direction` seems relevant only for NumberInput, perhaps we'd rather document any "extra" properties on a per component basis.

Reply:
Makes sense - Moved explanation of such extra properties to `<NumberInput>`.